pkg/reconciler/v1alpha1/taskrun/taskrun_test.go

```diff
@@ -732,16 +733,61 @@ func TestReconcileBuildUpdateStatus(t *testing.T) {
 		t.Fatalf("-want, +got: %v", d)
 	}
 
-	// update build status and trigger reconcile
+	// update build status and trigger reconcile : build is running
+	startTime := metav1.NewTime(time.Date(2018, time.November, 10, 23, 0, 0, 0, time.UTC))
+	build.Status.StartTime = &startTime
+	build.Status.Cluster = &buildv1alpha1.ClusterSpec{
+		Namespace: "default",
+		PodName:   "im-am-the-pod",
+	}
+	waiting := corev1.ContainerState{
+		Waiting: &corev1.ContainerStateWaiting{
+			Message: "foo",
+			Reason:  "bar",
+		},
+	}
+	build.Status.StepStates = []corev1.ContainerState{waiting}
+
+	if _, err = clients.Build.BuildV1alpha1().Builds(taskRun.Namespace).Update(build); err != nil {
+		t.Errorf("Unexpected error while creating build: %v", err)
+	}
+	if err := c.Reconciler.Reconcile(context.Background(), fmt.Sprintf("%s/%s", taskRun.Namespace, taskRun.Name)); err != nil {
+		t.Fatalf("Unexpected error when Reconcile(): %v", err)
+	}
+
+	newTr, err = clients.Pipeline.PipelineV1alpha1().TaskRuns(taskRun.Namespace).Get(taskRun.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Unexpected error fetching taskrun: %v", err)
+	}
+	if d := cmp.Diff(newTr.Status.PodName, build.Status.Cluster.PodName); d != "" {
+		t.Fatalf("-want, +got: %v", d)
+	}
+	if d := cmp.Diff(newTr.Status.StartTime, build.Status.StartTime); d != "" {
+		t.Fatalf("-want, +got: %v", d)
+	}
+	if d := cmp.Diff(newTr.Status.Steps, []v1alpha1.StepState{
+		{ContainerState: *waiting.DeepCopy()},
+	}); d != "" {
+		t.Fatalf("-want, +got: %v", d)
+	}
+	if d := cmp.Diff(newTr.Status.GetCondition(duckv1alpha1.ConditionSucceeded), buildSt, ignoreLastTransitionTime); d != "" {
+		t.Fatalf("-want, +got: %v", d)
+	}
```

Comment:
instead of testing this logic here, id really prefer to see the status logic broken out into a separate function and tested in isolation - this test is getting very complicated! the more complicated this test gets, the harder its going to be to maintain and change over time